Source/WebCore/page/EventHandler.h

```diff
@@ -383,9 +383,13 @@ class EventHandler {
     bool handleMousePressEventDoubleClick(const MouseEventWithHitTestResults&);
     bool handleMousePressEventTripleClick(const MouseEventWithHitTestResults&);
 
-    KeyboardScrollingAnimator* keyboardScrollingAnimatorForFocusedNode();
-    bool startKeyboardScrolling(KeyboardEvent&);
+    bool beginKeyboardScrollGesture(KeyboardScrollingAnimator*, KeyboardEvent&);
     void stopKeyboardScrolling();
+    bool startKeyboardScrollingOnFrame(KeyboardEvent&);
+    bool keyboardScrollOnRenderBoxLayer(KeyboardEvent&, RenderBox*);
```

Comment:
This and the next three functions all sound very similar, but some of them look at the ancestor chain and some do not; the names should reflect that. Also, there's existing code in `findEnclosingScrollableContainer` that is quite similar; perhaps we can share code, or at least share naming.

Also, I'd like the names to make it clear if these are just scrolling by one increment, or whether then are starting an entire scroll animation.